src/react/components/StyleForm.jsx

```diff
@@ -0,0 +1,348 @@
+         * @param {String} targetString - target parameter in object provided in full dot notation 
+         * @param {String|Number} value - value to be set
+         */
+        this.parseThru = (targetObject, targetString, value) => {
+            if (typeof targetString === 'string') {
+                return this.parseThru(targetObject, targetString.split('.'), value); // first cycle of recursion converts targetString into array
+            } else if (targetString.length == 1 && value !== undefined) {
+                return targetObject[targetString[0]] = value; // We reach end of the recursion
+            } else if (targetString.length === 0) {
+                return targetObject; // target is already on level 0 so no need for recursion
+            } else {
+                return this.parseThru(targetObject[targetString[0]], targetString.slice(1), value); // recursive call and remove first element from array
+            }
+        }
+    }
+
+    _getCurrentTab (tab) {
+        switch(tab) {
+            case 'point':
+                return (
+                    <PointTab
+                        formLayout={ formLayout }
+                        onChangeCallback={ (key, value) => this.updateState(key, value) }
+                        markers={ this.props.markers }
+                        styleSettings={ this.state }
+                    />
+                );
+            case 'line':
+                return (
+                    <LineTab
+                        formLayout={ formLayout }
+                        onChangeCallback={ (key, value) => this.updateState(key, value) }
+                        lineIcons={ lineIcons }
+                        styleSettings={ this.state }
+                    />
+                );
+            case 'area':
+                return (
+                    <AreaTab
+                        lineIcons={ lineIcons.lineDash }
+                        areaFills={ areaFills }
+                        formLayout={ formLayout }
+                        onChangeCallback={ (key, value) => this.updateState(key, value) }
+                        styleSettings={ this.state }
+                    />
+                );
+            default:
+                return false;
+        }
+    }
+
+    /**
+     * @method _createPreview
+     * @description Compose Preview -component
+     * @returns {React.Component} Preview component with provided style
+     */
+    _createPreview () {
+        return (
+            <Row>
+                <Preview
+                    markers={ this.props.markers }
+                    fillPatterns={ areaFills }
+                    styleSettings={ this.state }
+                />
+            </Row>
+        );
+    }
+
+    /**
+     * @method _getColorPickers
+     * @description Compose fill and stroke color picker elements for style form
+     * 
+     * @returns {React.Component} Only stroke color picker or both stroke and fill 
+     */
+    _getColorPickers () {
```

Comment:
Each form can init their own color pickers so it shouldn't be done on the main form level

Reply:
Changed so.